frame/staking/src/lib.rs

```diff
@@ -635,6 +736,22 @@ pub trait Trait: frame_system::Trait {
 
 	/// The NPoS reward curve to use.
 	type RewardCurve: Get<&'static PiecewiseLinear<'static>>;
+
+	/// Something that can predict the next session change.
+	type NextSessionChange: EstimateNextSessionChange<Self::BlockNumber>;
```

Comment:
Actually the type is sometimes fulfilled with something that can predict accurately (in mocks for example). So I disagree:  
`type NextSessionChange` is something that can predict (accurately or inaccurately) the next session change. 
I added to the docs of `EstimateNextSessionChange ` that : 
```
/// Something that can estimate at which block number the next era change will happen in a best
/// effort manner. The result may or may night be accurate, based on the chain configuration and
/// semantics.
pub trait EstimateNextSessionChange<BlockNumber>
```

Reply:
having in the doc of EstimateNextSessionChange that it can be an inaccurate estimation is fine to me.
But nonetheless the type NextSessionChange in staking can never be used for an accurate prediction of next session change (or it must be documented that this type must have an accurate prediction for some circumstances).
So EstimateNextSessionChange is still a better name to it for me.